### DISCUSSION TITLE
23 marzo 2023 - esercizio 3

### DISCUSSION THREAD

**Discussion by IonutCicio** [sapienzastudentsnetwork/basi-di-dati-1#37] [Esami]
Supponiamo di avere un file di 12.000.000 record. Ogni record occupa 275 byte, di cui 25 per il campo chiave.

Ogni blocco contiene 2048 byte. Un puntatore a blocco occupa 5 byte.

Usiamo una organizzazione B-tree in cui ogni blocco del file principale punta al prossimo blocco nel file principale (B+-tree).

I blocchi sia del file principale sono pieni al massimo mentre i blocchi del file indice sono pieni al minimo.

Calcolare:

- il numero di blocchi del file principale

- il numero di blocchi del file indice

- il numero di accessi necessari per ricercare un record del file principale

**Comment by niccolo-fato** [comment]
![IMAGE 2024-01-14 15:00:02](https://github.com/sapienzastudentsnetwork/bdi2324/assets/73995334/d396d0e4-503f-4577-abae-dd7fc1499d0d)

**Reply by IonutCicio** [reply]
Bisogna fare attenzione all'ultimo calcolo. Abbiamo 6 livelli in totale ma sono abbastanza sicuro che a lezione il prof. ha detto che bastano 5 accessi in questo caso, perché si suppone che la radice sia già in memoria principale.

**Comment by Warcophyr** [comment]
![image](https://github.com/sapienzastudentsnetwork/bdi2324/assets/133346488/1788984c-017f-4303-854b-56b99c747514)
![image](https://github.com/sapienzastudentsnetwork/bdi2324/assets/133346488/499f3218-b4d4-4033-9193-5d29e87a2f1c)
![image](https://github.com/sapienzastudentsnetwork/bdi2324/assets/133346488/bbb4f6a2-12b1-41b0-b024-c168ea87c0e2)

**Reply by Alby7503** [reply]
Nel calcolo di NKEY se ksize si riferisce alla dimensione della chiave allora dovrebbe essere 25

**Comment by sprigatita** [comment]
![IMG_0671](https://github.com/user-attachments/assets/ea9f07f5-a4c3-4114-a594-d5b24c3588e1)

**Comment by cek753** [comment]
<img width="1826" height="470" alt="image" src="https://github.com/user-attachments/assets/81990250-3d81-4c0b-870b-a83aae01730e" />
